src/taskgraph/transforms/chunking.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import copy
+from textwrap import dedent
+
+from voluptuous import ALLOW_EXTRA, Optional, Required
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import Schema
+from taskgraph.util.templates import substitute
+
+CHUNK_SCHEMA = Schema(
+    {
+        # Optional, so it can be used for a subset of jobs in a kind
+        Optional(
+            "chunk-config",
```

Comment:
I think I would mildly prefer this just be called `chunk` here (we don't append `-config` anywhere else). But not a big deal.

Reply:
That's a good call - this is the style we're using in `taskgraph` we may as well be consistent with it.